python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -338,9 +337,11 @@ def set_object_dtypes_from_pa_schema(df, schema):
     # "object" dtypes to agree with a specific
     # pyarrow schema.
     if schema:
-        for name in df.columns:
-            if name in schema.names and df[name].dtype == "O":
-                df[name] = df[name].astype(
+        for name in df._data.names:
```

Comment:
You can instead iterate over `df._data.items()` so that you don't have to index into `df._data` so many times below. You will still need it for the assignment target, but not in the `isinstance` check or the object of the astype call.

Reply:
Done